java_gen/pre-written/src/main/java/org/projectfloodlight/openflow/protocol/match/MatchField.java

```diff
@@ -31,15 +31,22 @@
 import org.projectfloodlight.openflow.types.VxlanNI;
 import org.projectfloodlight.openflow.types.VFI;
 
+import java.util.Set;
+import com.google.common.collect.ImmutableSet;
+
 public class MatchField<F extends OFValueType<F>> {
     private final String name;
     public final MatchFields id;
-    private final Prerequisite<?>[] prerequisites;
+    private final Set<Prerequisite<?>> prerequisites;
 
     private MatchField(final String name, final MatchFields id, Prerequisite<?>... prerequisites) {
         this.name = name;
         this.id = id;
-        this.prerequisites = prerequisites;
+        if (prerequisites == null) {
```

Comment:
I don't think you actually need the null check here, since `prerequisites` is a varargs parameter. If you pass no prerequisites, Java is smart enough to pass in a *zero-length* array, not a `null` reference.

The only way you could get a null reference for prerequisites if you *explicitly* passed in null as the the prerequisites array. Since the constructor is private, that could only happen from inside of this class.